server/buffer.c

```diff
@@ -300,6 +303,9 @@ static bool wait_for_buffer_ready(
      * with or until the deadline expires. */
     while (true)
     {
+        while (!buffer->released_once && !buffer->shutdown)
+            pwait_deadline(&buffer->mutex, &buffer->signal, &deadline);
```

Comment:
Hmmmm.  Think we'll need to talk about this, don't understand what this is all about, why it's needed.

Reply:
This is what is making the reader wait for the start event, otherwise the header will be sent before the start timestamp was captured

Reply:
Oh I see.  Hmm.  Not sure I like this.

So we have _two_ timeouts ticking away, one in the kernel interface, one in the buffer, but we're blocking the buffer timeout until the experiment has triggered.

Guess this is the right way.